### PR TITLE
Feature/3 components

### PR DESCRIPTION
**Pull request by timmi05**

### CODE REVIEW

File: src/result/panel/ResultSort.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+import Button from '../../component/Button'
+import './result_sort.css'
+
+const ResultSort = props => {
+
+    const { productSelected } = props;
+    const [disabled, setDisabled] = useState(false);
+
+    const handleSubmit1 = () => {
```
[the remark this answers is not in the document]
Comment (ant-plotnikov) [reply src/result/panel/ResultSort.jsx]: And tomorrow we can have the 3d and 4th button here, could you think of extracting handling logic to smth like `ButtonController` and make the `ResultSort` a simple presentational component?

File: prod.webpack.config.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ module.exports = {
       {
         test: /\.css$/,
         use: [MiniCssExtractPlugin.loader, 'css-loader'],
+        exclude: /node_modules/
```
[the remark this answers is not in the document]
Comment (timmi05) [reply prod.webpack.config.js]: Fixed

File: src/HeaderContainer.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                </span>
+                <span className='header-container-search'>
+                    < SearchLink
+                    productSelected = { productSelected }
+                    />
+                </span>
+            </div>
+            < Content
+                runtime = '145'
+                release_date = '2019'
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/HeaderContainer.jsx]: Fixed

File: src/app_container.css [hunk 4]
```diff
@@ -0,0 +1,13 @@
+.app-container{
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/app_container.css]: Fixed

File: src/component/SearchLink.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import './search_link.css';
+
+const SearchLink = props => {
+    const { productSelected } = props;
+
+    if ( productSelected ) {
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/component/SearchLink.jsx]: Fixed

File: src/component/SearchLink.jsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    return (
+            <a  className='search-link' href=''>search</a>
+        )
+    }
+    return (
+        <a/>
+    );
+};
+
+export default SearchLink
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/component/SearchLink.jsx]: Fixed

File: src/error/ErrorBoundary.jsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import AppContainer from '../AppContainer'
+
+export default class ErrorBoundary extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { hasError: false, error: null, errorInfo: null };
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/error/ErrorBoundary.jsx]: Fixed

File: src/index.html [hunk 8]
```diff
@@ -1,5 +1,9 @@
 <!DOCTYPE html>
 <html>
+<head>
+    <meta name="viewport">
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/index.html]: Fixed

File: src/product/DurationAndRelease.jsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import DigitalInfo from '../component/DigitalInfo';
+import './duration_and_release.css'
+
+const DurationAndRelease = props => {
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/product/DurationAndRelease.jsx]: Fixed

File: src/product/ProductContainer.jsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import ProductInfo from './ProductInfo';
+import './product_container.css';
+
+const ProductContainer = props => {
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/product/ProductContainer.jsx]: Fixed

File: src/result/ResultContainer.jsx [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import ResultPanel from './panel/ResultPanel';
+import ResultBody from './body/ResultBody';
+
+const ResultContainer = props => {
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/result/ResultContainer.jsx]: Fixed

File: src/result/body/ResultBody.jsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+import ResultItem from './ResultItem';
+import ResultList from './ResultList';
+import EmptyBody from './EmptyBody';
+import './result_body.css'
+
+const ResultBody = props => {
+    const { data } = props;
+
+   if (data.length > 0)
+    return (
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/result/body/ResultBody.jsx]: Fixed

File: src/result/panel/ResultInfo.jsx [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import './result_info.css'
+
+export const ResultInfo = props => {
+    const { productSelected, genre, data } = props;
+    const count = productSelected ? null : data.length;
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/result/panel/ResultInfo.jsx]: Fixed

File: src/result/panel/ResultInfo.jsx [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+    const massage = count && count !== 0 ? `${ count } movie found` : genre ? `Film by ${ genre } genre` : '';
```
[the remark this answers is not in the document]
Comment (timmi05) [reply src/result/panel/ResultInfo.jsx]: Fixed

File: src/result/panel/ResultSort.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (timmi05) [reply src/result/panel/ResultSort.jsx]: Discussed

**Pull request merged**